lib/ui/hash_codes.dart

```diff
@@ -28,95 +28,68 @@ int hashValues(
-            assert(arg07 is! Iterable);
-            result = 37 * result + arg07.hashCode;
-            if (arg08 != _hashEnd) {
-              assert(arg08 is! Iterable);
-              result = 37 * result + arg08.hashCode;
-              if (arg09 != _hashEnd) {
-                assert(arg09 is! Iterable);
-                result = 37 * result + arg09.hashCode;
-                if (arg10 != _hashEnd) {
-                  assert(arg10 is! Iterable);
-                  result = 37 * result + arg10.hashCode;
-                  if (arg11 != _hashEnd) {
-                    assert(arg11 is! Iterable);
-                    result = 37 * result + arg11.hashCode;
-                    if (arg12 != _hashEnd) {
-                      assert(arg12 is! Iterable);
-                      result = 37 * result + arg12.hashCode;
-                      if (arg13 != _hashEnd) {
-                        assert(arg13 is! Iterable);
-                        result = 37 * result + arg13.hashCode;
-                        if (arg14 != _hashEnd) {
-                          assert(arg14 is! Iterable);
-                          result = 37 * result + arg14.hashCode;
-                          if (arg15 != _hashEnd) {
-                            assert(arg15 is! Iterable);
-                            result = 37 * result + arg15.hashCode;
-                            if (arg16 != _hashEnd) {
-                              assert(arg16 is! Iterable);
-                              result = 37 * result + arg16.hashCode;
-                              if (arg17 != _hashEnd) {
-                                assert(arg17 is! Iterable);
-                                result = 37 * result + arg17.hashCode;
-                                if (arg18 != _hashEnd) {
-                                  assert(arg18 is! Iterable);
-                                  result = 37 * result + arg18.hashCode;
-                                  if (arg19 != _hashEnd) {
-                                    assert(arg19 is! Iterable);
-                                    result = 37 * result + arg19.hashCode;
-                                    if (arg20 != _hashEnd) {
-                                      assert(arg20 is! Iterable);
-                                      result = 37 * result + arg20.hashCode;
-                                      // I can see my house from here!
-                                    }
-                                  }
-                                }
-                              }
-                            }
-                          }
-                        }
-                      }
-                    }
-                  }
-                }
-              }
-            }
-          }
-        }
-      }
-    }
+  void append(Object o) {
+    assert(o is! Iterable);
+    result = 0x1fffffff & (result + o.hashCode);
+    result = 0x1fffffff & (result + ((0x0007ffff & result) << 10));
+    result = result ^ (result >> 6);
```

Comment:
It is Jenkins hash borrowed from an implementation in the sdk.
I had not got the "finish" operation right. It should be better now.